openedx/core/djangoapps/user_api/accounts/serializers.py

```diff
@@ -32,6 +38,111 @@ def get_identity(self, data):
             return None
 
 
+class UserProfileReadOnlySerializer(serializers.Serializer):
+    """
+    Class that serializes the portion of User model needed for account information.
+    """
+    def __init__(self, *args, **kwargs):
+        # Don't pass the 'configuration' arg up to the superclass
+        self.configuration = kwargs.pop('configuration', None)
+        if not self.configuration:
+            self.configuration = settings.ACCOUNT_VISIBILITY_CONFIGURATION
+
+        # Don't pass the 'custom_fields' arg up to the superclass
+        self.custom_fields = kwargs.pop('custom_fields', None)
+
+        super(UserProfileReadOnlySerializer, self).__init__(*args, **kwargs)
+
+    def to_native(self, user):
+        """
+        Overwrite to_native to handle custom logic since we are serializing two models as one here
+        :param user: User object
+        :return: Dict serialized account
+        """
+        profile = user.profile
+
+        data = {
```

Comment:
The current UserProfile serializer (`AccountLegacyProfileSerializer`) has that same shortcoming.  We must define which fields to serialize. I think it is actually a good thing to explicitly define attributes in a serializer, especially given how much massaging we need to do to the data 

```
class AccountLegacyProfileSerializer(serializers.HyperlinkedModelSerializer, ReadOnlyFieldsSerializerMixin):
    """
    Class that serializes the portion of UserProfile model needed for account information.
    """
    profile_image = serializers.SerializerMethodField("get_profile_image")
    requires_parental_consent = serializers.SerializerMethodField("get_requires_parental_consent")
    language_proficiencies = LanguageProficiencySerializer(many=True, allow_add_remove=True, required=False)

    class Meta(object):  # pylint: disable=missing-docstring
        model = UserProfile
        fields = (
            "name", "gender", "goals", "year_of_birth", "level_of_education", "country",
            "mailing_address", "bio", "profile_image", "requires_parental_consent", "language_proficiencies"
        )
        # Currently no read-only field, but keep this so view code doesn't need to know.
        read_only_fields = ()
        explicit_read_only_fields = ("profile_image", "requires_parental_consent")
```